### ISSUE TITLE
Add HTMLHyperlinkElementUtils

### ISSUE THREAD

**Issue by chexxor** [purescript-web/purescript-dom#143]
The `a` and `area` elements implement the HTMLHyperlinkElementUtils interface [0]. How are mixin interfaces like this usually included in this repo? It's not a proper element, just an extracted interface, so we can't make a new type in `DOM.HTML.Types` and check the tag name like this:

``` purescript
-- We can't do this, because an `a` element returns tag name of "HTMLAnchorElement".
readHTMLHyperlinkElementUtils :: Foreign -> F HTMLHyperlinkElementUtils
readHTMLHyperlinkElementUtils = unsafeReadTagged "HTMLHyperlinkElementUtils
```
But we could do this:

``` purescript
htmlAnchorElementToHTMLHyperlinkElementUtils :: HTMLAnchorElement -> HTMLHyperlinkElementUtils
htmlAnchorElementToHTMLHyperlinkElementUtils = unsafeCoerce
```

and put that interface's members in a `DOM.HTML.HTMLHyperlinkElementUtils` module, like the `href` function. Should we just do that?

[0] https://html.spec.whatwg.org/multipage/links.html#htmlhyperlinkelementutils

**Comment by garyb** [comment]
Yep, `unsafeCoerce` is fine here - it's not an upcast, so it won't fail, and since it's an interface there's no corresponding class we can `unsafeReadTagged` for.
